src/pencilkit.cs

```diff
@@ -0,0 +1,243 @@
+//
+// PencilKit C# bindings
+//
+// Authors:
+//	TJ Lambert  <t-anlamb@microsoft.com>
+//
+// Copyright 2019 Microsoft Corporation All rights reserved.
+//
+
+#if MONOMAC
+using AppKit;
+using UIColor = AppKit.NSColor;
+using UIImage = AppKit.NSImage;
+
+using UIScrollViewDelegate = Foundation.NSObjectProtocol;
+using UIScrollView = Foundation.NSObject;
+using UIGestureRecognizer = Foundation.NSObject;
+using UIResponder = Foundation.NSObject;
+using UIView = Foundation.NSObject;
+using UIWindow = Foundation.NSObject;
+#else
+using UIKit;
+#endif
+
+using System;
+using ObjCRuntime;
+using Foundation;
+using CoreGraphics;
+
+namespace PencilKit {
+
+	[iOS (13, 0), NoMac]
+	[Native]
+	enum PKEraserType : long {
+		Vector,
+		Bitmap,
+	}
+
+	[iOS (13, 0), NoMac]
+	enum PKInkType {
+		[Field ("PKInkTypePen")]
+		Pen,
+
+		[Field ("PKInkTypePencil")]
+		Pencil,
+
+		[Field ("PKInkTypeMarker")]
+		Marker,
+	}
+
+	[iOS (13, 0), NoMac]
+	[BaseType (typeof (NSObject))]
+	[Model (AutoGeneratedName = true)] [Protocol]
+	interface PKCanvasViewDelegate : UIScrollViewDelegate {
+
+		[Export ("canvasViewDrawingDidChange:")]
+		void DrawingDidChange (PKCanvasView canvasView);
```

Comment:
Nit: should this be DrawingChanged? Or do we try and keep the C# name as close to the ObjC name as possible?

Reply:
How close to ObjC we want to be is a subjective decision: in general we try to keep close to ObjC unless it makes sense not to (usually because it makes it easier to map between ObjC and C# for customers: like reading documentation, samples, stackoverflow, etc. If we name things differently, it can become quite hard to write the C# code if you know the particular ObjC API).

However, we do change things. For instance: we try to not add redundant information (thus `DrawingDidChange` and not `CanvasViewDrawingDidChange`, since the type is a `*CanvasView*` of some sort). Another typical case is that we move the first parameter name to the actual parameter (for instance `drawWithFoo:` would become `Draw (object foo)`).